### PR TITLE
Release process documentation

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Initial documentation around our release process

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs/docsite/rst/collection_release.rst

##### ADDITIONAL INFORMATION

It's vaguely deliberate that this isn't directly linked to from the rest of the documentation yet.  Relatively few people will be in a position to *use* this.

### CODE REVIEW

File: docs/docsite/rst/collection_release.rst [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+drives which versions of Python the collections support.
+
+SDK dependencies need to be updated in a number of places, primarily:
+
+* README.md
+* constraints.txt files (for our tests)
+* ``ansible_collections.amazon.aws.plugins.module_utils.botocore.MINIMUM_BOTOCORE_VERSION``
+* ``ansible_collections.amazon.aws.plugins.module_utils.botocore.MINIMUM_BOTO3_VERSION``
+
+The pull request to update the SDK requirements can also include dropping explicit requirments for a
```
Comment (alinabuzachis) [inline docs/docsite/rst/collection_release.rst:77]:
```suggestion
The pull request to update the SDK requirements can also include dropping explicit requirements for a
```

File: docs/docsite/rst/collection_release.rst [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+Releases can be announced using the GitHub:
+
+* `amazon.aws <https://github.com/ansible-collections/amazon.aws/releases>`_
+* `community.aws <https://github.com/ansible-collections/community.aws/releases>`_
+
+Update the topic
+----------------
+
+Once the release is available from Ansible Galaxy, the topic shoulc be updated in the
```
Comment (alinabuzachis) [inline docs/docsite/rst/collection_release.rst:298]:
```suggestion
Once the release is available from Ansible Galaxy, the topic should be updated in the
```

**Review by alinabuzachis — COMMENTED** [review]
Shouldn't we also mention to update AMAZON_AWS_COLLECTION_VERSION = "7.0.0-dev0" in this section? https://github.com/ansible-collections/amazon.aws/pull/1345/files#diff-bb0641b561153e04ec907c013789aba6abcbfd975b5dc779cfcb146eac1a8df5R115

File: docs/docsite/rst/collection_release.rst [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+#. Generating the release
+
+Preparing the branches involves creating a new ``stable`` branch, updating documentation links, and
+bumping the version for the ``main`` branch.
+
+Generating the release involves updating the version information, creating the changelog and
+tagging the release.  This part of the process is identical to performing :ref:`Minor and Patch releases`
+and will be covered in that section.
+
+Preflight checks
```
Comment (jillr) [inline docs/docsite/rst/collection_release.rst:41]:
```suggestion
Pre-flight checks
```

File: docs/docsite/rst/collection_release.rst [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+  ...
+
+**plugins/module_utils/common.py:**
+
+.. code-block:: python
+   AMAZON_AWS_COLLECTION_VERSION = "6.0.0"
+
+.. note::
+
+  Separately commiting the each of the changes to your local branch as you go will save you time if
```
Comment (jillr) [inline docs/docsite/rst/collection_release.rst:201]:
```suggestion
  Separately committing each of the changes to your local branch as you go will save you time if
```

File: docs/docsite/rst/collection_release.rst [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+All tests for the PR should pass prior to merging.  This pull request can be approved and merged as
+usual.
+
+Because ``CHANGELOG.rst`` is actually generated from ``changelogs/changelog.yaml``, if you need to
+fix issues with a changelog fragment, the easiest option is to revert the final changelog
+generation, fix the original fragment, and re-generate the final changelog (This is why you should
+commit small changes as you go).
+
+.. note::
+  Releases for amazon.aws should either be prepared by someone from the Anisble Cloud Content
```
Comment (jillr) [inline docs/docsite/rst/collection_release.rst:262]:
```suggestion
  Releases for amazon.aws should either be prepared by someone from the Ansible Cloud Content
```

File: docs/docsite/rst/collection_release.rst [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+
+  git fetch origin
+  git checkout stable-6
+  git reset --hard origin/stable-6
+  git tag 6.0.0
+  git push origin 6.0.0
+
+
+Announce the release
+====================
```
Comment (jillr) [inline docs/docsite/rst/collection_release.rst:287]: This section  should mention the bullhorn

File: docs/docsite/rst/collection_release.rst [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+
+Create a GitHub "Release"
+-------------------------
+
+Releases can be announced using the GitHub:
```
Comment (jillr) [inline docs/docsite/rst/collection_release.rst:292]: Apologies, I'm not sure what you're saying here.  Given the links I think you might mean "the person doing the release should draft a release note on the repo"?

File: docs/docsite/rst/collection_release.rst [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+
+.. code-block:: bash
+
+  git fetch origin --tags
+  git checkout main
+  git reset --hard origin/main
+  git checkout -b release/6.0.0/changelog-to-main
+  git cherry-pick -n 6.0.0
+  git checkout origin/main galaxy.yml
+  git commit -m "Add changelogs from release 6.0.0 to main
```
Comment (markuman) [inline docs/docsite/rst/collection_release.rst:336]:
```suggestion
  git commit -m "Add changelogs from release 6.0.0 to main"
```

**Pull request merged**